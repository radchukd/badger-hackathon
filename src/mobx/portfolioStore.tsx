import { action, makeAutoObservable } from 'mobx';

export type Vault = { id: number; value: string; tags: Array<string> };

export class PortfolioStore {
  public selectedVault?: Vault;
  public vaults: Array<Array<Vault>> = [
    [
      { id: 0, value: 'wBTC/Digg', tags: ['UNI'] },
      { id: 1, value: 'Badger/wBTC', tags: ['UNI'] },
    ],
    [
      { id: 2, value: 'Wrapped BTC/Digg', tags: ['SUSHI'] },
      { id: 3, value: 'Wrapped BTC/Badger', tags: ['SUSHI'] },
      { id: 4, value: 'Wrapped BTC/Wrapped Ether', tags: ['SUSHI'] },
    ],
    [
      { id: 5, value: 'crvRenWBTC', tags: ['CURVE'] },
      { id: 6, value: 'renBTC/wBTC/sBTC', tags: ['CURVE'] },
      { id: 7, value: 'tBTC/sBTCCrv LP', tags: ['CURVE'] },
      { id: 8, value: 'crvRenWBTC', tags: ['CURVE', 'HARVEST'] },
    ],
  ];
  public selectedVaults: Array<number> = this.vaults.flat().map((vault) => vault.id);

  constructor() {
    makeAutoObservable(this);
  }

  setSelectedVault = action((vault: string | undefined): void => {
    if (vault === undefined) {
      this.selectedVault = undefined;
    } else {
      const normalizedName = vault.replace('Sushiswap ', '').replace('Convex ', '');
      this.selectedVault = this.vaults.flat().find((v) => v.value === normalizedName);
    }
  });

  get areAllSelected(): boolean {
    return this.vaults.flat().length === this.selectedVaults.length;
  }

  isVaultSelected = (vault: Vault): boolean => {
    return this.selectedVaults.includes(vault.id);
  };

  onAllClick = action((): void => {
    if (this.areAllSelected) this.selectedVaults = [];
    else this.selectedVaults = this.vaults.flat().map((vault) => vault.id);
  });

  onVaultClick = action((vault: Vault): void => {
    this.selectedVaults = this.isVaultSelected(vault)
      ? this.selectedVaults.filter((selectedVaultId) => selectedVaultId !== vault.id)
      : [...this.selectedVaults, vault.id];
  });
}
