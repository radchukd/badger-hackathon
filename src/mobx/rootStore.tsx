import { RouterStore } from 'mobx-router';
import React from 'react';

import { AccountStore } from './accountStore';
import { PortfolioStore } from './portfolioStore';

export class RootStore {
  public router: RouterStore<RootStore>;

  constructor() {
    this.router = new RouterStore<RootStore>(this);
  }
}

export const rootStore = new RootStore();

export const accountStore = new AccountStore({ preload: true });

export const portfolioStore = new PortfolioStore();

export interface IStoreContext {
  rootStore: RootStore;
  accountStore: AccountStore;
  portfolioStore: PortfolioStore;
}

export const StoreContext = React.createContext({} as IStoreContext);

export const StoreProvider = StoreContext.Provider;
