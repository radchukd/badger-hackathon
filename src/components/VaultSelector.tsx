import { observer } from 'mobx-react-lite';
import React from 'react';

import { Box, Checkbox, FormControlLabel, IconButton, makeStyles } from '@material-ui/core';
import { Close } from '@material-ui/icons';

import { Vault } from '../mobx/portfolioStore';
import { StoreContext } from '../mobx/rootStore';
import useCardStyles from '../styles/cardStyles';
import useGlobalStyles from '../styles/globalStyles';
import VaultListItem from './VaultListItem';

const useStyles = makeStyles((theme) => ({
  dropdownContainer: {
    padding: theme.spacing(2),
    backgroundColor: theme.palette.background.paper,
    borderRadius: theme.spacing(1),
  },
  closeIcon: {
    color: theme.palette.primary.light,
  },
}));

export interface IVaultSelector {
  onClose: () => void;
}

const VaultSelector = observer(({ onClose }: IVaultSelector) => {
  const classes = { ...useStyles(), ...useCardStyles(), ...useGlobalStyles() };
  const store = React.useContext(StoreContext);
  const {
    portfolioStore: { vaults, areAllSelected, onAllClick, onVaultClick, isVaultSelected },
  } = store;

  return (
    <Box className={classes.dropdownContainer} data-testid="vault-selector-container">
      <Box display="flex" justifyContent="space-between">
        <Box ml={1}>
          <FormControlLabel
            control={
              <Checkbox
                name="allSettVaultsCheckbox"
                checked={areAllSelected}
                onClick={onAllClick}
                data-testid="vault-selector-vault-all-checkbox"
              />
            }
            label="All Sett Vaults"
          />
        </Box>
        <IconButton size="small" onClick={onClose}>
          <Close fontSize="small" className={classes.closeIcon} />
        </IconButton>
      </Box>
      <Box display="flex" flexWrap="wrap" className={classes.boxGap}>
        {vaults.map((subvaults, subVaultIndex) => (
          <Box key={subVaultIndex} display="flex" flexDirection="column" flexGrow={1}>
            {subvaults.map((vault, vaultIndex) => (
              <VaultListItem
                key={vaultIndex}
                vault={vault}
                isButton={true}
                onItemClick={(vault: Vault) => onVaultClick(vault)}
                isItemChecked={isVaultSelected(vault)}
                data-testid="vault-selector-vault-list-item"
              />
            ))}
          </Box>
        ))}
      </Box>
    </Box>
  );
});

export default VaultSelector;
