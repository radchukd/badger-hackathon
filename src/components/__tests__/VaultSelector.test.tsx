import '@testing-library/jest-dom';

import React from 'react';

import { render } from '@testing-library/react';

import { AccountStore } from '../../mobx/accountStore';
import { PortfolioStore } from '../../mobx/portfolioStore';
import { RootStore, StoreProvider } from '../../mobx/rootStore';
import VaultSelector from '../VaultSelector';

describe('VaultSelector', () => {
  const rootStore = new RootStore();
  const accountStore = new AccountStore({ preload: false });
  const portfolioStore = new PortfolioStore();
  const store = { rootStore, accountStore, portfolioStore };

  it('renders correctly', () => {
    const tree = render(
      <StoreProvider value={store}>
        <VaultSelector onClose={() => null} />
      </StoreProvider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
