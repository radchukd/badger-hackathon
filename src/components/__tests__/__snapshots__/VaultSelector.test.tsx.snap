// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaultSelector renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-17 makeStyles-dropdownContainer-1"
        data-testid="vault-selector-container"
      >
        <div
          class="MuiBox-root MuiBox-root-18"
        >
          <div
            class="MuiBox-root MuiBox-root-19"
          >
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-20 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-21 Mui-checked MuiIconButton-colorSecondary"
                data-testid="vault-selector-vault-all-checkbox"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-23"
                    data-indeterminate="false"
                    name="allSettVaultsCheckbox"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                All Sett Vaults
              </span>
            </label>
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-closeIcon-2 MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root MuiBox-root-24 makeStyles-boxGap-13"
        >
          <div
            class="MuiBox-root MuiBox-root-25"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
              data-testid="vault-selector-vault-list-item"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root makeStyles-vaultIcon-28"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root MuiBox-root-30"
              >
                wBTC/Digg
              </div>
              <div
                class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  UNI
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
              data-testid="vault-selector-vault-list-item"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root makeStyles-vaultIcon-28"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root MuiBox-root-31"
              >
                Badger/wBTC
              </div>
              <div
                class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  UNI
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-32"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
              data-testid="vault-selector-vault-list-item"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root makeStyles-vaultIcon-28"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root MuiBox-root-33"
              >
                Wrapped BTC/Digg
              </div>
              <div
                class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  SUSHI
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
              data-testid="vault-selector-vault-list-item"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root makeStyles-vaultIcon-28"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root MuiBox-root-34"
              >
                Wrapped BTC/Badger
              </div>
              <div
                class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  SUSHI
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
              data-testid="vault-selector-vault-list-item"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root makeStyles-vaultIcon-28"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root MuiBox-root-35"
              >
                Wrapped BTC/Wrapped Ether
              </div>
              <div
                class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  SUSHI
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-36"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
              data-testid="vault-selector-vault-list-item"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root makeStyles-vaultIcon-28"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root MuiBox-root-37"
              >
                crvRenWBTC
              </div>
              <div
                class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  CURVE
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
              data-testid="vault-selector-vault-list-item"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root makeStyles-vaultIcon-28"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root MuiBox-root-38"
              >
                renBTC/wBTC/sBTC
              </div>
              <div
                class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  CURVE
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
              data-testid="vault-selector-vault-list-item"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root makeStyles-vaultIcon-28"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root MuiBox-root-39"
              >
                tBTC/sBTCCrv LP
              </div>
              <div
                class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  CURVE
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
              data-testid="vault-selector-vault-list-item"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root makeStyles-vaultIcon-28"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root MuiBox-root-40"
              >
                crvRenWBTC
              </div>
              <div
                class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  CURVE
                </span>
              </div>
              <div
                class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  HARVEST
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-17 makeStyles-dropdownContainer-1"
      data-testid="vault-selector-container"
    >
      <div
        class="MuiBox-root MuiBox-root-18"
      >
        <div
          class="MuiBox-root MuiBox-root-19"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-20 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-21 Mui-checked MuiIconButton-colorSecondary"
              data-testid="vault-selector-vault-all-checkbox"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-23"
                  data-indeterminate="false"
                  name="allSettVaultsCheckbox"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              All Sett Vaults
            </span>
          </label>
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-closeIcon-2 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root MuiBox-root-24 makeStyles-boxGap-13"
      >
        <div
          class="MuiBox-root MuiBox-root-25"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
            data-testid="vault-selector-vault-list-item"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root makeStyles-vaultIcon-28"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="MuiBox-root MuiBox-root-30"
            >
              wBTC/Digg
            </div>
            <div
              class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                UNI
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
            data-testid="vault-selector-vault-list-item"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root makeStyles-vaultIcon-28"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="MuiBox-root MuiBox-root-31"
            >
              Badger/wBTC
            </div>
            <div
              class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                UNI
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-32"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
            data-testid="vault-selector-vault-list-item"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root makeStyles-vaultIcon-28"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="MuiBox-root MuiBox-root-33"
            >
              Wrapped BTC/Digg
            </div>
            <div
              class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                SUSHI
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
            data-testid="vault-selector-vault-list-item"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root makeStyles-vaultIcon-28"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="MuiBox-root MuiBox-root-34"
            >
              Wrapped BTC/Badger
            </div>
            <div
              class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                SUSHI
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
            data-testid="vault-selector-vault-list-item"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root makeStyles-vaultIcon-28"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="MuiBox-root MuiBox-root-35"
            >
              Wrapped BTC/Wrapped Ether
            </div>
            <div
              class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                SUSHI
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-36"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
            data-testid="vault-selector-vault-list-item"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root makeStyles-vaultIcon-28"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="MuiBox-root MuiBox-root-37"
            >
              crvRenWBTC
            </div>
            <div
              class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                CURVE
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
            data-testid="vault-selector-vault-list-item"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root makeStyles-vaultIcon-28"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="MuiBox-root MuiBox-root-38"
            >
              renBTC/wBTC/sBTC
            </div>
            <div
              class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                CURVE
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
            data-testid="vault-selector-vault-list-item"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root makeStyles-vaultIcon-28"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="MuiBox-root MuiBox-root-39"
            >
              tBTC/sBTCCrv LP
            </div>
            <div
              class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                CURVE
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-vaultItem-26 MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selectedVaultItem-27"
            data-testid="vault-selector-vault-list-item"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root makeStyles-vaultIcon-28"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="MuiBox-root MuiBox-root-40"
            >
              crvRenWBTC
            </div>
            <div
              class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                CURVE
              </span>
            </div>
            <div
              class="MuiChip-root makeStyles-vaultChip-29 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                HARVEST
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
